File: frontend/src/fakeBackend/services/EmissionTypeService.js
import { db } from "../db";
import { generateId } from "../util/generateID";

export const EmissionTypeService = {
  getEmissionType(emissionTypeId) {
    return db.emissionTypes.find(et => et.id === emissionTypeId);
  },

  getAllEmissionTypes() {
    return db.emissionTypes;
  },

  addEmissionType({ name, unit, desc, scope }) {
    const emissionType = {
      id: generateId(),
      name,
      unit,
      scope,
      desc
    };
    db.emissionTypes.push(emissionType);
    return emissionType;
  }
};

File: backend/src/main/java/com/emission_impossible/carbon_modelling_tool/service/EmissionTypeService.java
package com.emission_impossible.carbon_modelling_tool.service;

import com.emission_impossible.carbon_modelling_tool.model.EmissionType;
import com.emission_impossible.carbon_modelling_tool.repository.EmissionTypeRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class EmissionTypeService {

    private final EmissionTypeRepository emissionTypeRepository;

    public EmissionTypeService(EmissionTypeRepository emissionTypeRepository){
        this.emissionTypeRepository = emissionTypeRepository;
    }

    public EmissionType getEmissionType(Long emissionTypeId){

        return this.emissionTypeRepository.findById(emissionTypeId)
                .orElseThrow(() -> new RuntimeException(String.format("No emission type found for id %d", emissionTypeId)));
    }

    public List<EmissionType> getAllEmissionTypes(){
        return this.emissionTypeRepository.findAll();
    }

    public EmissionType addEmissionType(String name, String unit, String scope, String description){
        EmissionType emissionType = new EmissionType(name, unit, scope, description);

        this.emissionTypeRepository.save(emissionType);

        return emissionType;
    }
}
